build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -842,6 +842,58 @@ message NodeProperties {
 
   // The UNIX file mode, e.g., 0755.
   google.protobuf.UInt32Value unix_mode = 3;
+
+  // Constrains server-side unification of identical filesystem nodes.
+  NodeUnificationConstraint.Value unification_constraint = 4;
```

Comment:
Food for thought: `repeated OptimizationConstraint`? I'm not sure the field needs to be super specific that it's about NodeUnification, or could be expanded to cover other things later.